src/megaclient.cpp

```diff
@@ -14816,163 +14489,43 @@ void MegaClient::failSync(Sync* sync, SyncError syncerror)
     syncactivity = true;
 }
 
-void MegaClient::disableSync(Sync* sync, SyncError syncError)
-{
-    sync->errorCode = syncError;
-    sync->changestate(SYNC_DISABLED, syncError); //This will cause the later deletion of Sync (not MegaSyncPrivate) object
-
-    syncactivity = true;
-}
 
 bool MegaClient::disableSyncContainingNode(mega::handle nodeHandle, SyncError syncError)
 {
     auto sync = getSyncContainingNodeHandle(nodeHandle);
     if (sync)
     {
-        disableSync(sync, syncError);
-        return true;
+        syncs.disableSelectedSyncs([&](SyncConfig&, Sync* s) {
+            return s == sync;
+        }, syncError);
     }
     return false;
 }
 
 Sync * MegaClient::getSyncContainingNodeHandle(mega::handle nodeHandle)
 {
-    while(!ISUNDEF(nodeHandle))
+    Sync* syncFound = nullptr;
+    while(!ISUNDEF(nodeHandle) && !syncFound)
     {
-        for (sync_list::iterator it = syncs.begin(); it != syncs.end(); it++)
-        {
-            Sync *sync = (*it);
+        syncs.forEachRunningSync([&](Sync* sync) {
+
             if (sync->localroot && sync->localroot->node && sync->localroot->node->nodehandle == nodeHandle)
             {
-                return sync;  // no nested synched folders is allowed, so if found, there's no more active syncssyncs
+                syncFound =  sync;  // no nested synched folders is allowed, so if found, there's no more active syncssyncs
             }
-        }
+        });
 
         Node *n = nodebyhandle(nodeHandle);
         nodeHandle = n ? n->parenthandle : UNDEF;
     }
-    return nullptr;
+    return syncFound;
 }
 
 void MegaClient::failSyncs(SyncError syncError)
```

Comment:
I wonder why aren't you moving this one to Syncs too?

Reply:
I didn't need to change it, so I thought it would make review easier if it's left where it is for now, as the PR is already very large.

Reply:
ok, it seems odd to have those there, but we can live with that ;-)